### PR TITLE
ocaml-libvirt: 0.6.1.4.2017-11-08-unstable -> 0.6.1.5

### PR DESCRIPTION
**Pull request by quyse**
###### Motivation for this change

Fix a few problems and bumps it to the latest version:

* `git://git.annexia.org/git/ocaml-libvirt.git` was not working, it should have been `git://git.annexia.org/ocaml-libvirt.git`
* Anyway http://git.annexia.org/?p=ocaml-libvirt.git;a=summary says the repo is obsolete, suggesting https://libvirt.org/git/?p=libvirt-ocaml.git;a=summary which in turn is a mirror of https://gitlab.com/libvirt/libvirt-ocaml.git which seems to be a right source to use
* Comments from git log suggest other packages needed 0.6.1.5 version while it wasn't even released. It is released now, so let's use it.
* Plus build fixes for new version.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/ocaml-libvirt/default.nix [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
+  version = "0.6.1.5";
 
-  src = fetchgit {
-    url = "git://git.annexia.org/git/ocaml-libvirt.git";
-    rev = rev;
-    sha256 = "0vxgx1n58fp4qmly6i5zxiacr7303127d6j78a295xin1p3a8xcw";
+  src = fetchFromGitLab {
+    owner = "libvirt";
+    repo = "libvirt-ocaml";
+    rev = "refs/tags/v${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/ocaml-modules/ocaml-libvirt/default.nix:10]:
```suggestion
    rev = "v${version}";
```

Shouldn't this just be the long of my change?

Comment (quyse) [reply pkgs/development/ocaml-modules/ocaml-libvirt/default.nix]: Thanks, I didn't realize it works for tags with `fetchFromGitlab`.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102533` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>virt-top</li>
  </ul>
</details>
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>libguestfs</li>
    <li>python27Packages.guestfs</li>
    <li>python37Packages.guestfs</li>
    <li>python38Packages.guestfs</li>
    <li>vagrant</li>
  </ul>
</details>

**Comment by vbgl** [comment]
What about using the sources from there: <https://libvirt.org/sources/ocaml/ocaml-libvirt-0.6.1.5.tar.gz>?

**Comment by quyse** [comment]
It seems fix for `virt-top` with `ocaml-libvirt-0.6.1.5` was committed immediately after release of the `virt-top` latest version: http://git.annexia.org/?p=virt-top.git;a=summary
I included it as a patch for `virt-top` - hope it's OK to do this in this PR as it is related.

Another small change for `ocaml-libvirt`: made use of `autoreconfHook` which I guess is preferred.

> What about using the sources from there: https://libvirt.org/sources/ocaml/ocaml-libvirt-0.6.1.5.tar.gz?

I'm fairly new to contributing to nixpkgs, so I don't know - whatever is usually preferred (is there some policy on using git vs tarballs, or Github/Gitlab vs project websites?) Normally I prefer Github/Gitlab for any sources because they are less likely to go down or change domain or other part of URL.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102533` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>python27Packages.guestfs</li>
    <li>python37Packages.guestfs</li>
    <li>python38Packages.guestfs</li>
  </ul>
</details>

**Comment by vbgl** [comment]
@GrahamcOfBorg build libguestfs vagrant virt-top

**Pull request merged**